include/grpcpp/security/tls_certificate_provider.h

```diff
@@ -36,6 +40,11 @@ class GRPCXX_DLL CertificateProviderInterface {
  public:
   virtual ~CertificateProviderInterface() = default;
   virtual grpc_tls_certificate_provider* c_provider() = 0;
+
+  // Returns an OK status if the credentials held by the provider are valid.
+  // What it means for a credential to be valid is determined by the provider
```

Comment:
This statement seems so weak that I'm not sure the API actually has value.  I think we probably need to define this a bit more rigorously in order for it to be useful.

Reply:
Removed from the base class.